File: jvm/src/main/java/org/karabalin/s0877/Solution0877.java
package org.karabalin.s0877;

import org.junit.jupiter.api.Assertions;

import java.util.Arrays;

class Solution {
    public boolean stoneGame(int[] piles) {
        Arrays.sort(piles);
        int a = 0, b = 0;
        for (int i = 0; i < piles.length - 1; ++i) {
            b += piles[i++];
            a += piles[i];
        }
        return a > b;
    }
}

public class Solution0877 {
    public static void main(String[] args) {
        var s = new Solution();
        Assertions.assertTrue(s.stoneGame(new int[]{5, 3, 4, 5}));
    }
}

File: kotlin/src/main/kotlin/org/karabalin/s0877/Solution0877.kt
package org.karabalin.s0877

class Solution {
    fun stoneGame(piles: IntArray): Boolean {
        piles.sort()
        var a = 0
        var b = 0

        var i = 0
        while (i < piles.size - 1) {
            b += piles[i++]
            a += piles[i++]
        }

        return a > b
    }
}

fun main() {
    val solution = Solution()
    println(solution.stoneGame(intArrayOf(5, 3, 4, 5)))
}
